src/test-integration/java/fi/fmi/avi/archiver/populator/FailingPopulatorTest.java

```diff
@@ -0,0 +1,130 @@
+package fi.fmi.avi.archiver.populator;
+
+import fi.fmi.avi.archiver.AviationMessageArchiver;
+import fi.fmi.avi.archiver.TestConfig;
+import fi.fmi.avi.archiver.database.DatabaseAccess;
+import fi.fmi.avi.archiver.initializing.AviationProductsHolder;
+import fi.fmi.avi.archiver.message.ArchiveAviationMessage;
+import fi.fmi.avi.archiver.message.populator.MessagePopulator;
+import fi.fmi.avi.model.GenericAviationWeatherMessage;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.ConfigDataApplicationContextInitializer;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest({"auto.startup=false"})
+@Sql(scripts = {"classpath:/schema-h2.sql", "classpath:/h2-data/avidb_message_types_test.sql",
+        "classpath:/h2-data/avidb_message_format_test.sql", "classpath:/h2-data/avidb_message_routes_test.sql",
+        "classpath:/h2-data/avidb_stations_test.sql"}, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+@Sql(scripts = "classpath:/h2-data/avidb_cleanup_test.sql", executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+@ContextConfiguration(classes = {AviationMessageArchiver.class, TestConfig.class,
+        FailingPopulatorTest.FailingPopulatorConfig.class},//
+        loader = AnnotationConfigContextLoader.class,//
+        initializers = {ConfigDataApplicationContextInitializer.class})
+public class FailingPopulatorTest {
+
+    private static final int WAIT_MILLIS = 100;
+    private static final int TIMEOUT_MILLIS = 1000;
+    private static final File BASE_DIR = new File(System.getProperty("java.io.tmpdir") + "/.avi-message-archiver");
```

Comment:
Good idea in general. These integration tests use the configuration in [application.yml](https://github.com/fmidev/aviation-message-archiver/blob/master/src/test-integration/resources/application.yml) so just changing the paths in the test code does not work. But it might be worth it to work on a solution.

Reply:
Of course! That's why it is the same with `AviationMessageArchiverTest`. But it works for now, and we've got better things to do, so I will not create a new issue on this. We can return to this if it becomes a problem or a trivial solution suddenly just appears in front of our eyes.